pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -514,21 +561,32 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 		Context("when the MAPI machine does not exist and the CAPI machine does", func() {
 			Context("and there is no CAPI machineSet owning the machine", func() {
 				BeforeEach(func() {
-					By("Creating the CAPI machine")
-					capiMachine = capiMachineBuilder.Build()
-					Eventually(k8sClient.Create(ctx, capiMachine)).Should(Succeed(), "capi machine should be able to be created")
 
-					By("Updating the CAPA machine adding the CAPI machine as an owner")
-					Eventually(k.Update(capaMachine, func() {
-						capaMachine.OwnerReferences = append(capaMachine.OwnerReferences, metav1.OwnerReference{
-							Kind:               machineKind,
-							APIVersion:         clusterv1.GroupVersion.String(),
-							Name:               capiMachine.Name,
-							UID:                capiMachine.UID,
-							BlockOwnerDeletion: ptr.To(true),
-							Controller:         ptr.To(false),
-						})
-					})).Should(Succeed(), "capa machine should be able to be updated")
+					// The CAPI Machine must reference the CAPA machine,
+					// so build the CAPA machine (without the owner reference) fist.
+					// Note we don't create it on the API server yet.
+					noMachineSetCapaMachineBuilder := capaMachineBuilder.WithName("test-machine-no-machineset")
+					capaMachine = noMachineSetCapaMachineBuilder.Build()
+
+					capiMachine = capiMachineBuilder.WithName("test-machine-no-machineset").WithInfrastructureRef(corev1.ObjectReference{
+						Kind:      capaMachine.Kind,
+						Name:      capaMachine.GetName(),
+						Namespace: capaMachine.GetNamespace(),
+					}).Build()
+					Expect(k8sClient.Create(ctx, capiMachine)).Should(Succeed())
```

Comment:
I'd rather use Eventuallys instead of Expect all over the place for these k8s api calls